providers/apis/defi/raydium/multi_client.go

```diff
@@ -0,0 +1,116 @@
+package raydium
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"go.uber.org/zap"
+
+	oracleconfig "github.com/skip-mev/slinky/oracle/config"
+)
+
+// MultiJSONRPCClient is an implementation of the SolanaJSONRPCClient interface that delegates
+// requests to multiple underlying clients, and aggregates over all provided responses.
+type MultiJSONRPCClient struct {
+	// underlying clients
+	clients []SolanaJSONRPCClient
+
+	// logger
+	logger *zap.Logger
+}
+
+func NewMultiJSONRPCClient(clients []SolanaJSONRPCClient, logger *zap.Logger) *MultiJSONRPCClient {
+	return &MultiJSONRPCClient{
+		clients: clients,
+		logger:  logger,
+	}
+}
+
+// NewMultiJSONRPCClientFromEndpoints creates a new MultiJSONRPCClient from a list of endpoints.
+func NewMultiJSONRPCClientFromEndpoints(endpoints []oracleconfig.Endpoint, logger *zap.Logger) *MultiJSONRPCClient {
+	clients := make([]SolanaJSONRPCClient, len(endpoints))
+	for i := range endpoints {
+		client := rpc.New(endpoints[i].URL)
+		clients[i] = client
+	}
+	return NewMultiJSONRPCClient(clients, logger)
+}
+
+// GetMultipleAccountsWithOpts delegates the request to all underlying clients and applies a filter
+// to the responses.
+func (c *MultiJSONRPCClient) GetMultipleAccountsWithOpts(
```

Comment:
There is only one implementation of this function.

Can we just call it `GetMultipleAccounts`?

Reply:
The interface is defined in accordance so that the default client generated stubs are an implementation, otherwise we'd have to make a wrapper / not use the getMultipleAccountsWithOpts method on the client